frontend/app/services/timezone-service.js

```diff
@@ -44,18 +44,27 @@ module.exports = function(ConfigurationService, I18n) {
       return d;
     },
 
+    parseDate: function(date, format) {
+      return moment(date, format);
+    },
+
     parseISODate: function(date) {
       return TimezoneService.parseDate(date, 'YYYY-MM-DD');
     },
 
     formattedDate: function(date) {
-      var format = ConfigurationService.dateFormatPresent() ? ConfigurationService.dateFormat() : 'L';
-      return TimezoneService.parseDate(date).format(format);
+      var d = TimezoneService.parseDate(date);
+      return d.format(TimezoneService.getDateFormat());
     },
 
     formattedTime: function(date) {
-      var format = ConfigurationService.timeFormatPresent() ? ConfigurationService.timeFormat() : 'LT';
-      return TimezoneService.parseDate(date).format(format);
+      return TimezoneService.parseDatetime(date).format(TimezoneService.getTimeFormat());
+    },
+
+    formattedDatetime: function(date) {
```

Comment:
I *personally* would go as far as to call them `dateString` and `datetimeString`, because we need to parse them first...